### PR TITLE
Implement the actions-client for EdgeGuardian

### PR DESCRIPTION
**Pull request by jvimal-eg**
- Add the yaml file
- Implement the runner in main.js, shell.js, cleanup.js
- Check in the compiled dist/ files

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const core = require('@actions/core');
+const { shell } = require('./shell.js');
+
+async function install() {
+    await shell(`
+    curl --proto '=https' --tlsv1.2 -sSf https://edgeguard-app.s3.us-west-1.amazonaws.com/linux/install.sh | sudo bash
```
Comment (vadim-eg) [inline main.js:6]: Not sure that this is the best way to translate shell usage to js, but, whatever works.

File: main.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    await shell(`egctl advanced token-login ${token}`)
+}
+
+async function status() {
+    await shell(`egctl status`)
+}
+
+async function main() {
+    const token = core.getInput('api_key');
+    await install();
```
Comment (vadim-eg) [inline main.js:20]: As install requires sudo - it may make sense to be able to use existing pre-provisioned client - e.g. check for runnable egctl .

**Review by vadim-eg — APPROVED** [review]

File: main.js (hunk 2 above)
Comment (jvimal-eg) [reply main.js:20]: Good idea, I will address this in the next iteration.  Keeping what works as is for now.

**Comment by jvimal-eg** [comment]
Hmm, since @vadim-eg approved it without me requesting for approval, GitHub is not allowing me to merge this PR.  So, @theron-eg you'll need to take another look;-)

**Review by vadim-eg — APPROVED** [review]

**Comment by vadim-eg** [comment]
@jvimal-eg you can probably try re-requesting.

**Review by vadim-eg — APPROVED** [review]
Take #3

**Comment by vadim-eg** [comment]
Looks like it doesn't care if I approve or not. Happens all the time ,)

**Comment by theron-eg** [comment]
The key is this "1 approving review by reviewers with **_write access_**".  I added Vadim with write access and now the PR can be merged.

**Comment by vadim-eg** [comment]
https://media.giphy.com/media/3xz2BKqajaSGpBLmec/giphy.gif

**Comment by jvimal-eg** [comment]
Aha thanks folks!

**Pull request merged**